### PR TITLE
bank migrated to guice

### PR DESCRIPTION
**Pull request by kristiyanpetkov**
In my my PR i migrated my bank using Guice Dependency Injection.

### CODE REVIEW

File: task5/src/main/java/com/clouway/adapter/http/MyGuiceServletConfig.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package com.clouway.adapter.http;
```
Comment (panayotkulchev) [inline task5/src/main/java/com/clouway/adapter/http/MyGuiceServletConfig.java]: could you offer me a better packages for this?

File: task5/src/main/java/com/clouway/adapter/http/MyServletModule.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import com.google.inject.servlet.ServletModule;
+
+/**
+ * Created by Kristiyan Petkov  <kristiqn.l.petkov@gmail.com> on 06.07.16.
+ */
+public class MyServletModule extends ServletModule {
+  @Override
+  protected void configureServlets() {
+    filter("/*").through(ConnectionFilter.class);
+    filter("/useraccout", "/history", "/bankoperationhandler").through(SecurityFilter.class);
```
Comment (panayotkulchev) [inline task5/src/main/java/com/clouway/adapter/http/MyServletModule.java]: What will happen if you have 100 pages to filter... you will have very long list here. It is better to filter `*` and exclude the once that you do not want to filter.

File: task5/src/main/java/com/clouway/adapter/http/MyServletModule.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    filter("/login").through(LoginFilter.class);
+
+    serve("/login").with(LoginPage.class);
+    serve("/bankoperationhandler").with(BankOperationHandler.class);
+    serve("/register").with(RegisterPage.class);
+    serve("/history").with(TransactionHistory.class);
+    serve("/useraccount").with(UserAccount.class);
+    serve("/logincontroller").with(LoginController.class);
+    serve("/registercontroller").with(RegisterController.class);
```
Comment (panayotkulchev) [inline task5/src/main/java/com/clouway/adapter/http/MyServletModule.java]: When you use this line, is it needet to use `@WebServlet(name = "RegisterController")` in the servlet

File: task5/src/main/java/com/clouway/core/ApplicationModule.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+ * Created by Kristiyan Petkov  <kristiqn.l.petkov@gmail.com> on 06.07.16.
+ */
+public class ApplicationModule extends AbstractModule {
+  @Override
+  protected void configure() {
+    install(new PersistenceModule());
+  }
+
+  @Provides
+  public ConnectionProvider getConnection() {
```
Comment (panayotkulchev) [inline task5/src/main/java/com/clouway/core/ApplicationModule.java]: may be `PersistenceModule` is better place

File: task5/src/main/java/com/clouway/adapter/http/MyGuiceServletConfig.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import com.clouway.core.ApplicationModule;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.servlet.GuiceServletContextListener;
+import com.google.inject.servlet.ServletModule;
+
+/**
+ * Created by Kristiyan Petkov  <kristiqn.l.petkov@gmail.com> on 06.07.16.
+ */
+public class MyGuiceServletConfig extends GuiceServletContextListener {
```
Comment (panayotkulchev) [inline task5/src/main/java/com/clouway/adapter/http/MyGuiceServletConfig.java]: `AppConfig` is enought. Here is will be the main config of your app. `MyGuiceServletConfig` sound good name for tutorials

**Comment by panayotkulchev** [comment]
Good job. Only couple small notes!

File: task5/src/main/java/com/clouway/adapter/http/MyGuiceServletConfig.java (hunk 1 above)
Comment (kristiyanpetkov) [inline task5/src/main/java/com/clouway/adapter/http/MyGuiceServletConfig.java]: yes, i think `package com.clouway.server` is a better package for the AppConfig.

File: task5/src/main/java/com/clouway/adapter/http/MyServletModule.java (hunk 2 above)
Comment (kristiyanpetkov) [inline task5/src/main/java/com/clouway/adapter/http/MyServletModule.java]: You are right. I made it to filter all requests using `filter("/*")` and in the Security filter I exclude the once that I don't want to filter.

File: task5/src/main/java/com/clouway/adapter/http/MyServletModule.java (hunk 3 above)
Comment (kristiyanpetkov) [inline task5/src/main/java/com/clouway/adapter/http/MyServletModule.java]: Absolutely not. All `@WebServlet` annotations removed.

File: task5/src/main/java/com/clouway/core/ApplicationModule.java (hunk 4 above)
Comment (kristiyanpetkov) [inline task5/src/main/java/com/clouway/core/ApplicationModule.java]: Yes, it is.  Moved.

File: task5/src/main/java/com/clouway/adapter/http/MyGuiceServletConfig.java (hunk 5 above)
Comment (kristiyanpetkov) [inline task5/src/main/java/com/clouway/adapter/http/MyGuiceServletConfig.java]: Renamed.

**Comment by panayotkulchev** [comment]
LGTM!

**Pull request closed without merging**